GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -968,7 +972,7 @@ public void PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> get
                     }
                 }).FileAndForget();
 
-                cancellationToken.ThrowIfCancellationRequested();
+                // TODO: can never have an effect in combination with _isRefreshingRevisions: cancellationToken.ThrowIfCancellationRequested();
```

Comment:
I think this may be a detrimental change. Instead, I think we should do this:
```
                cancellationToken.Register(() => _isRefreshingRevisions = false);
```

https://docs.microsoft.com/dotnet/api/system.threading.cancellationtoken.register

Reply:
> I think this may be a detrimental change. Instead, I think we should do this:
```
                cancellationToken.Register(() => _isRefreshingRevisions = false);
```
🚀 This is an easy way to avoid `_isRefreshingRevisions` remaining `true`.

Though we must basically await the cancellation of the previous async tasks before we start the next updates.
We have already noticed this necessity in tests of Forms.
This could be done by using dedicated instances of `JoinableTaskFactory`.

---

The structure of this function is:
```
if (_isRefreshingRevisions) return;
_isRefreshingRevisions = true;
_refreshRevisionsSequence.Next();
do some sync stuff which never requests cancellation
RunAsync(...).FileAndForget();
cancellationToken.ThrowIfCancellationRequested();
```
The cancellation is requested by the async code (on completion / on error). So it cannot have been triggered here yet.

I just wanted to clarify how the code works and to reduce the confusion for future readers about this pointless check for cancellation. Or have I missed anything?

Reply:
This is probably not the only pointless check for cancellation...
The cancellation could occur early in the async code (not likely)

The solution is not obvious.
The cancellationToken is needed only in the try{}, but this (minor) change could be merged.

Reply:
> The cancellation could occur early in the async code (not likely)

Yes, and if the async task is executed by a background thread, it can already trigger the cancellation before this sync function finishes (on the UI thread).
So this check for cancellation should not be removed.